### PR TITLE
Feat/generators

### PR DESCRIPTION
**Pull request by ra7kar**
Generator example

### CODE REVIEW

File: data_structures/generator.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# generator example
+
+# first we will implement a regular function to generated Factors. The we will
+# implement the a generator of Factors.
+
+"""regular function to generate factors.
+def factors(k):
```
Comment (atrifex) [inline data_structures/generator.py:7]: Delete commented out code.

Comment (atrifex) [inline data_structures/generator.py:7]: Delete commented out code.

Comment (ra7kar) [reply data_structures/generator.py]: Deleted commented code.

Comment (ra7kar) [reply data_structures/generator.py]: Deleted commented code

**Pull request merged**